lib/plug/ip_whitelist/environment_variable_parser.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Plug.IpWhitelist.EnvironmentVariableParser do
+  @moduledoc """
+    IP Whitelists often need to be stored as an environment variable.
+    This module provides functionality for parsing a list of IP ranges out of a
+    string which could be stored in your application's environment variables.
+    Example:
+      "1.1.1.1-1.1.1.1 2.3.4.5-6.7.8.9"
+  """
+
+  @doc """
+    Given a string of ip ranges, convert to a list of ip ranges
+    Example input:
+      "1.1.1.1-1.1.1.1 2.3.4.5-6.7.8.9"
+    Example output:
+      [
+        {{1, 1, 1, 1}, {1, 1, 1, 1}},
+        {{2, 3, 4, 5}, {6, 7, 8, 9}}
+      ]
+  """
+  @spec parse(String.t()) :: [IpWhitelist.ip_range(), ...]
+  def parse(string_ip_whitelist) do
+    string_ip_whitelist
+    |> String.split(" ")
+    |> Enum.map(&parse_ip_range/1)
+  end
+
+  @doc """
+    Given a string which contains an ip range, convert to an ip_range
+    Example input:
+      1.1.1.1-1.1.1.1
+    Example output:
+      {{1, 1, 1, 1}, {1, 1, 1, 1}}
+  """
+  @spec parse_ip_range(String.t()) :: IpWhitelist.ip_range()
+  def parse_ip_range(string_ip_range) do
```

Comment:
@nixpulvis 
I didn't know what CIDR was so I looked it up: https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing
I'm totally on board with the idea of using a predefined format for ip ranges instead of the weird syntax I came up with here.
But I'm a little confused about the CIDR syntax and its potential limitations.
Based on the wikipedia article alone, it seems like its impossible to define very specific IP ranges such as 0.0.0.3-0.0.0.155 for example. 
How would you define that range in CIDR?

Reply:
Yea, I'm not sure CIDR fits our exact requirements. I think plenty of tools allow both `<ip>-<ip>` ranges, and `ip/<cidr>` net classes (`nmap` for example). So I'm guessing this just falls in the "fun fact category".